### PR TITLE
Support augmentation advanced settings

### LINKED ISSUE #699: Enable data augmentation across all Geti tasks
**User story**

To comply with data governance regulations, some users need insight into the data augmentation methods that are applied to their data in Geti for training. In addition, customers requested these augmentation settings to be configurable, so that they can increase the diversity of their dataset and therefore the robustness of their trained model. 

The objective of this requirement is to finalise the data augmentation implementation: expose configurable data augmentation parameters as part of the advanced model training flow for all applicable tasks (excluding classification, as that has been accommodated as part of the August release).

**Requirements**
- Configurable data augmentation options shall be exposed across all applicable tasks in Geti (classification has already been accommodated).

- All configurable parameters and their values shall be available to users as a single overview for each trained model version.

- Internal and external documentation shall be updated, at minimum with an overview of this Data section of the advanced model training flow, as well as the intended use of the configurable parameters & their purpose.

### LINKED ISSUE #978: Introduce configurable augmentations for all supervised tasks
Geti 2.12 exposed augmentation parameters for the classification task; the goal now is to do the same for all other tasks

### PR DESCRIPTION
**Pull request by romanowska**
## 📝 Description

Support augmentation advanced settings in model training
Fix bug of lack of display of advanced settings mode

Screenshot 2025-09-04 at 14 58 49

## ✨ Type of Change

Select the type of change your PR introduces:

- [x] 🐞 **Bug fix** – Non-breaking change which fixes an issue
- [x] 🚀 **New feature** – Non-breaking change which adds functionality
- [ ] 🔨 **Refactor** – Non-breaking change which refactors the code base
- [x] 💥 **Breaking change** – Changes that break existing functionality
- [ ] 📚 **Documentation update**
- [ ] 🔒 **Security update**
- [ ] 🧪 **Tests**

## 🧪 Testing Scenarios

Describe how the changes were tested and how reviewers can test them too:

- [x] ✅ Tested manually
- [ ] 🤖 Run automated end-to-end tests

## ✅ Checklist

Before submitting the PR, ensure the following:

- [x] 🔍 PR title is clear and meaningful
- [x] ✍️ PR description clearly explains the changes and their reason
- [x] 📝 I have linked the PR to the corresponding GitHub Issues, if any
- [x] 💬 I have commented my code, especially in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ✅ I have added tests that prove my fix is effective or my feature works

### CODE REVIEW

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            onChange([start, end]);
+        }
+    };
+
+    useEffect(() => {
+        setParameterValue({
+            start: value[0],
+            end: value[1],
+        });
+    }, [value]);
```
Comment (MarkRedeman) [inline web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx]: np: I'd move this to line 39,
```
    const [parameterValue, setParameterValue] = useState<RangeValue<number>>({
        start: value[0],
        end: value[1],
    });
```
This way it is more obvious that we want to synchronize the `paramterValue` every time that `value` changes.

**Review by MarkRedeman — DISMISSED** [review]

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    onChange,
+    isDisabled,
+    name,
+    step,
+}) => {
+    const [parameterValue, setParameterValue] = useState<RangeValue<number>>({
+        start: value[0],
+        end: value[1],
+    });
+    const fieldStep = getStep({ step, maxValue: value[1], minValue: value[0] });
```
[the remark this answers is not in the document]
Comment (romanowska) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx]: updated

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx (hunk 1 above)
Comment (romanowska) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx]: updated

**Review by MarkRedeman — DISMISSED** [review]

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    isDisabled,
+    name,
+    step,
+}) => {
+    const [parameterValue, setParameterValue] = useState<RangeValue<number>>({
+        start: value[0],
+        end: value[1],
+    });
+
+    useEffect(() => {
```
Comment (jpggvilaca) [inline web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx:40]: do we need this `useEffect`? we are already explicitly calling setState and onChange whenever necessary

**Review by jpggvilaca — COMMENTED** [review]
Could you add a few unit tests for this new component? It has a ton of edge cases

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx (hunk 3 above)
Comment (romanowska) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx:40]: we do it for when we set range parameters values to default for RangeParameterField to reflect that change

**Review by MarkRedeman — APPROVED** [review]

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx (hunk 3 above)
Comment (dwesolow) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx:40]: When this could happen? When user changes parameters, then in real time we update `parameterValue` and only `onChangeEnd` we update `value` but at that time `parameterValue` should be equal to `value`. Isn't it the case?

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        let { start, end } = inputValue;
+        // Prevent start and end from being equal
+        if (end - start > fieldStep) {
+            setParameterValue({ start, end });
+        }
+    };
+
+    const handleNumberChange = (start: number, end: number): void => {
+        // Prevent start and end from being equal
+        if (end - start > fieldStep) {
```
[the remark this answers is not in the document]
Comment (dwesolow) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx:64]: I believe it's correct.

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        }
+    };
+
+    return (
+        <Flex gap={'size-100'}>
+            <NumberField
+                isQuiet
+                step={fieldStep}
+                value={parameterValue.start}
+                minValue={defaultValue[0]}
```
Comment (dwesolow) [inline web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx:76]: Just to double check, are we sure that `defaultValue` should be used as a `minValue` and `maxValue`? Will `defaultValue` change (on a server side) when we update `value`?

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.test.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    const defaultValue = [0.5, 1.5];
+    const name = 'Scaling ratio range';
+    const onChange = jest.fn();
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    const renderApp = ({ value = defaultValue, isDisabled = false }: { value?: number[]; isDisabled?: boolean }) => {
+        return render(
```
Comment (dwesolow) [inline web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.test.tsx:20]: np. we do have `providersRender` that uses `RequiredProviders`

**Comment by dwesolow** [comment]
Could you please add component and/or unit tests that verifies those augmentation parameters?

**Review by jpggvilaca — APPROVED** [review]

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx (hunk 5 above)
Comment (romanowska) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx:76]: yes, positive.
`defaultValue` never changes and returns the maximum range for the parameter.
when we update `value` only `value` changes and `defaultValue` stays default.

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.test.tsx (hunk 6 above)
Comment (romanowska) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.test.tsx:20]: i'll update it in next PR

**Comment by romanowska** [comment]
> Could you please add component and/or unit tests that verifies those augmentation parameters?

yes, I'll add more tests in my next PR

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx (hunk 3 above)
Comment (romanowska) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/range-parameter-field.component.tsx:40]: for when we set to default value
